contracts/EasyAuction.sol

```diff
@@ -217,23 +230,42 @@ contract EasyAuction is Ownable {
         uint256 auctionId,
         uint96[] memory _minBuyAmounts,
         uint96[] memory _sellAmounts,
-        bytes32[] memory _prevSellOrders
+        bytes32[] memory _prevSellOrders,
+        bytes memory allowListCallData
     ) public atStageOrderPlacement(auctionId) returns (uint64 userId) {
         return
             _placeSellOrders(
                 auctionId,
                 _minBuyAmounts,
                 _sellAmounts,
-                _prevSellOrders
+                _prevSellOrders,
+                allowListCallData
             );
     }
 
     function _placeSellOrders(
         uint256 auctionId,
         uint96[] memory _minBuyAmounts,
         uint96[] memory _sellAmounts,
-        bytes32[] memory _prevSellOrders
+        bytes32[] memory _prevSellOrders,
+        bytes memory allowListCallData
     ) internal returns (uint64 userId) {
+        {
+            address allowListManger =
+                auctionDataAccessManager[auctionId].allowListManager;
+            if (allowListManger != address(0)) {
+                (bool success, ) =
+                    allowListManger.call(
+                        abi.encodeWithSelector(
+                            auctionDataAccessManager[auctionId]
+                                .allowListFunction,
+                            allowListCallData,
+                            msg.sender
+                        )
+                    );
```

Comment:
> single type of interaction

To clarify what this mean: if you want to have arbitrary logic for verification, then I recommend using EIP-1271.There is a standard `verify` function that you can call, and this can execute any smart contract logic including that in `AllowListOffChainManaged`.

Reply:
Fixed it with static calls